src/main/java/com/mojang/brigadier/context/CommandContextBuilder.java

```diff
@@ -67,6 +119,13 @@ public S getSource() {
         return this;
     }
```

Comment:
This method is in fact called in lib code. It is used in the parse -method of both `LiteralCommandNode` and `ArgumentCommandNode`
Also it does not "Overwrite" the parameters.

The `range` becomes the old range plus the new range. So if, in this particular CommandContext, the parsed range was 5, 10 and the given range is 10, 16, the new range will be 5, 16 as this is what was parsed in this context.

Both modifer and forks are always null/false (aka not set), unless it's the last node of a particular command. Attemting to add a node to a CommandNode with a redirect will throw an `IllegalStateException`, similar when trying to add a child node to a command node that has a redirect modifier.
Of course, since this is a public memeber, you could abuse this method in many ways, but it is really only intended to be used by `CommandNode#parse(StringReader, CommandContextBuilder)`. Maybe it should throw Exceptions if someone tries to do stupid stuff with it.

Reply:
Tbh, I don't know how useful the context builder in general is as a public member. Usually, when executing a command, you do not get access to the builder, only the context after building.
The only reason it's public is probably because it gets used by other packages internally. In fact, the only class that creates instances of this class is the CommandDispatcher.

Reply:
Yea, it would have been best as an internal class, but that's not gonna happen in java.

I guess I will throw a disclaimer at the class doc, that this is an internal class and all bets are off.
But I think I will leave the javadoc on its members, to aid people mangling with the library.

Reply:
Hm. It comes in handy if you add a new command node type. Not sure if doing that is expected though, or if it should only be done by the library itself.